src/StreamJsonRpc.Tests/JsonRpcOverloadTests.cs

```diff
@@ -0,0 +1,253 @@
+
+        this.clientRpc = new JsonRpcOverload(this.clientStream, this.serverStream, this.server);
+        this.clientRpc.StartListening();
+    }
+
+    [Fact]
+    public async Task CanInvokeMethodOnServer()
+    {
+        string testLine = "TestLine1" + new string('a', 1024 * 1024);
+        string result1 = await this.clientRpc.InvokeAsync<string>(nameof(Server.ServerMethod), testLine);
+        Assert.Equal(testLine + "!", result1);
+    }
+
+    [Fact]
+    public async Task CanInvokeMethodThatReturnsCancelledTask()
+    {
+        var ex = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => this.clientRpc.InvokeAsync(nameof(Server.ServerMethodThatReturnsCancelledTask)));
+        Assert.Equal(CancellationToken.None, ex.CancellationToken);
+    }
+
+    [Fact]
+    public async Task CannotPassExceptionFromServer()
+    {
+        OperationCanceledException exception = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => this.clientRpc.InvokeAsync(nameof(Server.MethodThatThrowsUnauthorizedAccessException)));
+        Assert.NotNull(exception.StackTrace);
+
+        await Assert.ThrowsAsync<ObjectDisposedException>(() => this.clientRpc.InvokeAsync<string>(nameof(Server.ServerMethod), "testing"));
+    }
+
+    [Fact]
+    public async Task CanCallAsyncMethodThatThrows()
+    {
+        Exception exception = await Assert.ThrowsAnyAsync<TaskCanceledException>(() => this.clientRpc.InvokeAsync<string>(nameof(Server.AsyncMethodThatThrows)));
+        Assert.NotNull(exception.StackTrace);
+
+        await Assert.ThrowsAnyAsync<ObjectDisposedException>(() => this.clientRpc.InvokeAsync<string>(nameof(Server.AsyncMethodThatThrows)));
+    }
+
+    [Fact]
+    public async Task ThrowsIfCannotFindMethod_Overload()
+    {
+        await Assert.ThrowsAsync(typeof(RemoteMethodNotFoundException), () => this.clientRpc.InvokeAsync("missingMethod", 50));
+
+        var result = await this.clientRpc.InvokeAsync<string>(nameof(Server.ServerMethod), "testing");
+        Assert.Equal("testing!", result);
+    }
+
+    // Covers bug https://github.com/Microsoft/vs-streamjsonrpc/issues/55
+    // Covers bug https://github.com/Microsoft/vs-streamjsonrpc/issues/56
+    [Fact]
+    public async Task InvokeWithCancellationAsync_CancelOnFirstWriteToStream()
+    {
+        // TODO: remove the next line when https://github.com/Microsoft/vs-threading/issues/185 is fixed
+        this.server.DelayAsyncMethodWithCancellation = true;
+
+        // Repeat 10 times because https://github.com/Microsoft/vs-streamjsonrpc/issues/56 is a timing issue and we may miss it on the first attempt.
+        for (int iteration = 0; iteration < 10; iteration++)
+        {
+            using (var cts = new CancellationTokenSource())
+            {
+                this.clientStream.BeforeWrite = (stream, buffer, offset, count) =>
+                {
+                    // Cancel on the first write, when the header is being written but the content is not yet.
```

Comment:
`content is not yet`? Is it me or does this sound like it's missing something or requires a different phrasing?